File: t-tasks/src/main/java/com/teo/ttasks/widget/TasksWidgetService.java
package com.teo.ttasks.widget;

import android.content.Intent;
import android.widget.RemoteViewsService;

import com.teo.ttasks.data.remote.TasksHelper;

import javax.inject.Inject;

import dagger.android.AndroidInjection;

public class TasksWidgetService extends RemoteViewsService {

    @Inject TasksHelper tasksHelper;

    @Override
    public RemoteViewsFactory onGetViewFactory(Intent intent) {
        AndroidInjection.inject(this);
        return new TasksRemoteViewsFactory(getApplicationContext(), intent, tasksHelper);
    }
}

File: t-tasks/src/main/java/com/teo/ttasks/widget/TasksWidgetService.kt
package com.teo.ttasks.widget

import android.content.Intent
import android.widget.RemoteViewsService
import com.teo.ttasks.data.remote.TasksHelper
import org.koin.android.ext.android.inject

class TasksWidgetService : RemoteViewsService() {

    private val tasksHelper: TasksHelper by inject()

    override fun onGetViewFactory(intent: Intent): RemoteViewsFactory {
        return TasksRemoteViewsFactory(applicationContext, intent, tasksHelper)
    }
}
